src/core/mac/sub_mac.cpp

```diff
@@ -1165,18 +1165,19 @@ void SubMac::HandleCslTimer(void)
         }
 
         Get<Radio>().UpdateCslSampleTime(mCslSampleTime.GetValue());
-        if (mState == kStateCslSample)
+
+        if (RadioSupportsReceiveTiming())
         {
-            if (RadioSupportsReceiveTiming())
+            if (mState != kStateDisabled && mCslChannel)
```

Comment:
The original code is used to ensure that the CSL sample won't break other ongoing submac operations. What will happen if the `ReceiveAt` schedules at a time that the previous transmission hasn't ended? With the original logic, that 'Receive' will be skipped. Will the current implementation do the same thing? @edmont

Reply:
That's actually the case I'm aiming to fix here, when the CSL timer happens to be triggered while the radio is performing a transmission operation. If the `ReceiveAt` is cancelled there won't be a reception slot scheduled in the future and the radio implementation won't be able to append proper CSL IEs in the subsequent transmissions, leading to CSL de-synchronization.

Notice that `OPENTHREAD_CONFIG_CSL_RECEIVE_TIME_AHEAD` can be tuned by the platform implementation in order to invoke `ReceiveAt` well in advance of the actual receive operation, so there is time for the current transmission operation to finish. And in any case it's responsibility of the radio platform to handle possible conflicts with the delayed operations.

Reply:
Yes, the transmission will complete. I've tested this change and it fixes the case in which CSL timer triggers when the child is performing transmissions (i.e.: sending a multi-fragment Echo Reply).

Reply:
Sounds good! Thanks!

Reply:
> Does this mean, even if, `ReceiveAt` triggers at the time a TX hasn't ended, the TX will still complete?

I don't think it's strictly required that radio platform implementation works like this. I believe right now it is up to the radio platform implementation (or underlying hardware) to decide if TX is aborted, or RX is skipped if such conflict occurs.

I think it's fine to leave the decision to radio platform. We don't need to enforce it to be functionally identical with OT MAC implementation.